bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/DFDDataFlowAnalysisBuilder.java

```diff
@@ -109,6 +124,12 @@ protected void validate() {
      */
     public DFDConfidentialityAnalysis build() {
         this.validate();
-        return new DFDConfidentialityAnalysis(this.getEffectiveResourceProvider(), this.pluginActivator, this.modelProjectName);
+        DFDResourceProvider ressourceProvider = this.getEffectiveResourceProvider();
+
+        if (customTransposeFlowGraphFinderClass == null)
+            return new DFDConfidentialityAnalysis(ressourceProvider, this.pluginActivator, this.modelProjectName);
+        else
+            return new DFDConfidentialityAnalysis(ressourceProvider, this.pluginActivator, this.modelProjectName,
+                    this.customTransposeFlowGraphFinderClass);
```

Comment:
Use `customTransposeFlowGraphFinder.orElse(new CyclicTransposeFlowGraphFinder(resourceProvider))` here to get the used transpose flow graph finder.